src/main/java/com/dongsoop/dongsoop/chat/controller/ChatController.java

```diff
@@ -33,65 +34,93 @@ public ResponseEntity<ChatRoom> createRoom(@RequestBody CreateRoomRequest reques
+        return ResponseEntity.ok(syncedMessages);
     }
 
     @PostMapping("/room/{roomId}/recreate")
     public ResponseEntity<ChatRoom> recreateRoom(
             @PathVariable("roomId") String roomId,
             @RequestBody MessageSyncRequest request) {
-        chatService.recreateRoomIfNeeded(roomId, getCurrentUserId(), request.getMessages());
-        return ResponseEntity.ok(chatService.getChatRoomById(roomId));
+        Long currentUserId = getCurrentUserId();
+        chatService.recreateRoomIfNeeded(roomId, currentUserId, request.getMessages());
+
+        ChatRoom recreatedRoom = chatService.getChatRoomById(roomId);
+        return ResponseEntity.ok(recreatedRoom);
     }
 
     @PostMapping("/room/group")
     public ResponseEntity<ChatRoom> createGroupRoom(@RequestBody CreateGroupRoomRequest request) {
         Long currentUserId = getCurrentUserId();
+        Set<Long> participantIds = convertNicknamesToIds(request.getParticipants());
 
-        Set<Long> participantIds = request.getParticipants().stream()
-                .map(nickname -> memberService.getLoginAuthenticateByNickname(nickname).getId())
-                .collect(Collectors.toSet());
-
-        return ResponseEntity.ok(chatService.createGroupChatRoom(currentUserId, participantIds));
+        ChatRoom groupRoom = chatService.createGroupChatRoom(currentUserId, participantIds, request.getTitle());
+        return ResponseEntity.ok(groupRoom);
     }
 
     @PostMapping("/room/{roomId}/kick")
     public ResponseEntity<ChatRoom> kickUserFromRoom(
             @PathVariable("roomId") String roomId,
             @RequestBody KickUserRequest kickUserRequest) {
         Long currentUserId = getCurrentUserId();
+        String userToKickNickname = kickUserRequest.getUserId();
+        Long userToKickId = memberService.getLoginAuthenticateByNickname(userToKickNickname).getId();
 
-        Long userToKickId = memberService.getLoginAuthenticateByNickname(kickUserRequest.getUserId()).getId();
-
-        return ResponseEntity.ok(chatService.kickUserFromRoom(roomId, currentUserId, userToKickId));
+        ChatRoom updatedRoom = chatService.kickUserFromRoom(roomId, currentUserId, userToKickId);
+        return ResponseEntity.ok(updatedRoom);
     }
 
     private Long getCurrentUserId() {
         Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
         return Long.parseLong(authentication.getName());
     }
+
+    private Set<Long> convertNicknamesToIds(Set<String> nicknames) {
+        Set<String> safeNicknames = getSafeNicknames(nicknames);
+        return safeNicknames.stream()
+                .map(this::convertNicknameToId)
+                .collect(Collectors.toSet());
+    }
+
+    private Set<String> getSafeNicknames(Set<String> nicknames) {
+        if (nicknames != null) {
+            return nicknames;
+        }
+        return Collections.emptySet();
```

Comment:
부정문보다 긍정문을 사용할 때 가독성이 극대화됩니다.
```suggestion
        if (nicknames == null) {
            return Collections.emptySet();
        }
        
        return nicknames;
```